3-networks-dual-svpc/envs/shared/remote_state.tf

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+data "terraform_remote_state" "bootstrap" {
+  backend = "gcs"
+
+  config = {
+    bucket = "${var.backend_bucket}"
+    prefix = "terraform/bootstrap/state"
+  }
+}
+
+data "terraform_remote_state" "org" {
+  backend = "gcs"
+
+  config = {
+    bucket = "${var.backend_bucket}"
+    prefix = "terraform/org/state"
+  }
+}
+
+data "terraform_remote_state" "env_development" {
+  backend = "gcs"
+
+  config = {
+    bucket = "${var.backend_bucket}"
+    prefix = "terraform/environments/development"
+  }
+}
+
+data "terraform_remote_state" "env_non_production" {
+  backend = "gcs"
+
+  config = {
+    bucket = "${var.backend_bucket}"
+    prefix = "terraform/environments/non-production"
+  }
+}
+
+data "terraform_remote_state" "env_production" {
+  backend = "gcs"
+
+  config = {
+    bucket = "${var.backend_bucket}"
+    prefix = "terraform/environments/production"
```

Comment:
As a follow up we should look at UX for adding/deleting stages. An approach could be a dynamic "env" which we source from 1-org but we can discuss further in an issue.